Sources/IdentityState.swift

```diff
@@ -95,6 +95,32 @@ class IdentityState {
         return hasBooted
     }
 
+    /// When the advertising identifier from the `event` is different from the current value, it updates the persisted value and creates
+    /// a new XDM shared state. A consent request event is dispatched when advertising tracking preferences change.
+    /// - Parameters:
+    ///   - event: event containing a new ADID value.
+    ///   - createXDMSharedState: function which creates new XDM shared state
+    ///   - eventDispatcher: function which dispatches events to the event hub
+    func updateAdvertisingIdentifier(event: Event,
+                                     createXDMSharedState: ([String: Any], Event) -> Void,
+                                     eventDispatcher: (Event) -> Void) {
+        
+        // update adid if changed and extract the new adid value
+        let (adIdChanged, shouldUpdateConsent) = shouldUpdateAdId(newAdID: event.adId)
+        if adIdChanged {
+            let adId = event.adId ?? ""
```

Comment:
Now that `Event.adID` is non-optional, you can remove the nil coalescing operator.